### PR TITLE
Allow independent scrolling in split-column mode

### PR DESCRIPTION
**Pull request by maskeynihal**
## Type of Change

- **Something else:**  Add independent scrolling

## What issue does this relate to?

resolves https://github.com/digitalocean/nginxconfig.io/issues/430

### What should this PR do?

- Add style `overflowY: auto` and `height: 100vh` for each column so that it can be scrolled horizontally. 

### What are the acceptance criteria?

- Independent scrolling should be allowed for each column on `Side-by-side` mode.

### Before

https://github.com/digitalocean/nginxconfig.io/assets/26411488/f0858305-6089-40f7-aeef-c2aba6f97eac

### After

https://github.com/digitalocean/nginxconfig.io/assets/26411488/c0d9eb34-388f-4ecb-855a-b67c487d6ef2

### CODE REVIEW

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
Please could we apply this styling in the SCSS, rather than mixing inline styles with classes?

**Comment by maskeynihal** [comment]
@MattIPv4 What should we do for "Back to top" button?

**Comment by MattIPv4** [comment]
> What should we do for "Back to top" button?

Good question -- what do you think would be the best UX? I'm thinking it should scroll the document, and both the columns, back to the top?

File: src/nginxconfig/scss/_columns.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/*
+Copyright 2022 DigitalOcean
```
Comment (MattIPv4) [inline src/nginxconfig/scss/_columns.scss:2]:
```suggestion
Copyright 2023 DigitalOcean
```

File: src/nginxconfig/templates/app.vue
Comment (MattIPv4) [inline src/nginxconfig/templates/app.vue:1]: Ditto in here!

**Comment by maskeynihal** [comment]
@MattIPv4 I have updated the copyright year to 2023. 

Should I update the copyright year on every file of the whole repo?

**Comment by maskeynihal** [comment]
> > What should we do for "Back to top" button?
> 
> Good question -- what do you think would be the best UX? I'm thinking it should scroll the document, and both the columns, back to the top?

Yes, that sounds right. Should I do that in this PR or in another PR?

**Comment by MattIPv4** [comment]
> Should I update the copyright year on every file of the whole repo?

Only the files you have touched in this PR, please

> Yes, that sounds right. Should I do that in this PR or in another PR?

It'd be great to do it in this PR, please, as it's related to the change being made here

**Comment by maskeynihal** [comment]
@MattIPv4 I have updated the code. Now, the independent column also scrolls when "Back to top" is clicked.

File: src/nginxconfig/templates/footer.vue
Comment (MattIPv4) [inline src/nginxconfig/templates/footer.vue:1]: Copyright year needs a bump pleaes

File: src/nginxconfig/templates/footer.vue [hunk 2]
```diff
@@ -68,5 +68,14 @@ THE SOFTWARE.
         components: {
             ExternalLink,
         },
+        methods: {
+            handleScrollToTop: () => {
+                window.scrollTo({top: 0});
+
+                const columns = document.querySelectorAll('.column-scroll-y');
+
+                columns.forEach(column => column.scrollTo({top: 0}));
```
Comment (MattIPv4) [inline src/nginxconfig/templates/footer.vue:77]:
```suggestion
                window.scrollTo({ top: 0 });
                document.querySelectorAll('.column-scroll-y')
                  .forEach(column => column.scrollTo({ top: 0 }));
```
Also a nitpick, formatting + removing the redundant const (lmk if eslint disagrees with this)

Comment (maskeynihal) [reply src/nginxconfig/templates/footer.vue]: Ya, you are right. I have removed redundant cosnt.

File: src/nginxconfig/templates/footer.vue [hunk 3]
```diff
@@ -28,7 +28,7 @@ THE SOFTWARE.
     <div class="footer">
         <div class="container">
             <p>
-                <a href="#top" class="button is-primary is-small">{{ $t('templates.footer.backToTop') }}</a>
+                <button type="button" class="button is-primary is-small" @click="handleScrollToTop">{{ $t('templates.footer.backToTop') }}</button>
```
[the remark this answers is not in the document]
Comment (maskeynihal) [reply src/nginxconfig/templates/footer.vue]: I thought the formatting will be handled by pre-commit hooks. I have updated it manually here. 

Should we add `prettier` here so pre-commit hooks handles these kind of formatting?

File: src/nginxconfig/templates/footer.vue
Comment (maskeynihal) [reply src/nginxconfig/templates/footer.vue:1]: Updated.

File: src/nginxconfig/templates/footer.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (maskeynihal) [reply src/nginxconfig/templates/footer.vue]: @MattIPv4 I have added prettier via this PR. https://github.com/digitalocean/nginxconfig.io/pull/448

Please do check

**Pull request merged**